legate/core/launcher.py

```diff
@@ -538,64 +670,77 @@ def update_storages(self):
 class TaskLauncher:
     def __init__(
         self,
-        context,
-        task_id,
-        mapper_id=0,
-        tag=0,
-        error_on_interference=True,
-        side_effect=False,
-    ):
+        context: Context,
+        task_id: int,
+        mapper_id: int = 0,
+        tag: int = 0,
+        error_on_interference: bool = True,
+        side_effect: bool = False,
+    ) -> None:
         assert type(tag) != bool
         self._context = context
         self._runtime = context.runtime
         self._core_types = self._runtime.core_context.type_system
         self._task_id = task_id
         self._mapper_id = mapper_id
-        self._inputs = []
-        self._outputs = []
-        self._reductions = []
-        self._scalars = []
-        self._comms = []
+        self._inputs: list[Union[RegionFieldArg, FutureStoreArg]] = []
```

Comment:
I feel there should be a protocol for these two classes, but that doesn't need to happen in this PR.

Reply:
https://github.com/nv-legate/legate.core/issues/192